src/net_processing.cpp

```diff
@@ -472,15 +472,16 @@ class PeerManagerImpl final : public PeerManager
         EXCLUSIVE_LOCKS_REQUIRED(!m_peer_mutex);
     void BlockChecked(const CBlock& block, const BlockValidationState& state) override
         EXCLUSIVE_LOCKS_REQUIRED(!m_peer_mutex);
-    void NewPoWValidBlock(const CBlockIndex *pindex, const std::shared_ptr<const CBlock>& pblock) override;
+    void NewPoWValidBlock(const CBlockIndex *pindex, const std::shared_ptr<const CBlock>& pblock) override
+        EXCLUSIVE_LOCKS_REQUIRED(!m_most_recent_block_mutex);
```

Comment:
I wonder how useful it is to annotate interface functions with member mutexes. The only allowed caller can't see the mutex, as it is private.

Reply:
An annotation is not for a caller, rather for Clang Thread Safety Analysis C++ extension.

Reply:
(An external caller (accessing via the virtual/override) can't see the mutex, but an in-class caller could)

Reply:
Yeah, I was just thinking that it would be incredibly unlikely or even wrong to even have an in-class caller.